wade-review.md

```diff
@@ -0,0 +1,596 @@
+        try (final FileInputStream input = new FileInputStream(fileName)) {
+            allProps.load(input);
+        }
+        return allProps;
+    }
+```
+
+I’ll load the properties – and hold on, let’s make the method that loads them. This method points at the path to our properties file and returns the contents.
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+}
+
+Let’s add a couple things to the properties down here– consumer group id, set the offset to earliest, and a max cache bytes config.
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+
+}
+
+I’ll call the  GitHubPrRatio method, to get the ratio coming in.
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+        final KafkaStreams streams = new KafkaStreams(gitHubPrRatio.topology(props), props);
+
+}
+
+Create a new Streams instance with our topology. What’s a topology? Hold on a sec, I’ll show you, but first, let’s tie this up.
+
+
+```java
+
+    public static void main(final String[] args) throws Exception {
+
+        final Properties props = GitHubPrRatio.loadEnvProperties("get-started.properties");
+
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, "streams-github-87");
+        props.put(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+
+        GitHubPrRatio gitHubPrRatio = new GitHubPrRatio();
+
+        final KafkaStreams streams = new KafkaStreams(gitHubPrRatio.topology(props), props);
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            streams.close(Duration.ofSeconds(5));
+            latch.countDown();
+        }));
+}
+
+Then we’ll set up a countdown latch so it waits for other threads before it starts…
```

Comment:
I'm a bit confused about what the latch is doing based on this explanation.

If I read the code, it seems to suggest that when it goes to "shutdown" it should wait for the latch.

But your script suggests the wait happens before it starts up.

Reply:
Also, I've never used a Countdown Latch before so I don't really know what it does regardless. So that doesn't help.

Reply:
https://github.com/Cerchie/notes-for-gh-video/pull/2/commits/dadfe630441c7a138602ba697f7ddec154d4b10f